Tools/Scripts/layout-test-worker-parser

```diff
@@ -237,22 +251,74 @@ def load_from_file(path):
 
 # Loads the stdio input for analysis from a build URL.
 # Returns a str with the contents of the stdio log.
-def load_from_buildbot(url):
-    url = url.replace('/#/', '/api/v2/') + ("/raw" if "/logs/stdio" in url else "/steps/layout-test/logs/stdio/raw")
-    try: # validation
-        uobj = urlparse(url)
+def load_from_url(base_url: str, step_name: str = None, no_warn: bool = False) -> str:
+    '''Takes a build URL and analyzes the steps to determine which log to return. Can also take a step stdio link ("/steps/<step-name>/logs/stdio") or a .txt file URL.'''
+
+    if not base_url.endswith('.txt'):
+        base_url = base_url.replace('/#/', '/api/v2/').strip('/')
+        # Don't analyze steps if user specifically asked for stdio
+        if '/logs/stdio' not in base_url:
+            base_url += '/steps'
+        elif not base_url.endswith('/raw'):
+            base_url += '/raw'
+
+    try:
+        uobj = urlparse(base_url)
         if not all([uobj.scheme, uobj.netloc]):
             raise IOError()
     except:
-        error("Invalid URL specified: %s.", url)
+        error(f'Invalid URL specified: {base_url}.')
+        graceful_exit(os.EX_DATAERR)
+
+    info('Loading from URL (this may take a moment)...')
+
+    try:
+        data = requests.get(base_url)
```

Comment:
Why are we loading the `base_url` instead of the `uobj` you just created? `base_url` usually means you're mutating the URL into some final form. Seems like `base_url` is really `constructed_url`, or something to that effect.

Reply:
Good point, I fixed this in the latest commit